server/ai_process.go

```diff
@@ -382,10 +400,77 @@ func submitUpscale(ctx context.Context, params aiRequestParams, sess *AISession,
 	return resp.JSON200, nil
 }
 
-func processAIRequest(ctx context.Context, params aiRequestParams, req interface{}) (*worker.ImageResponse, error) {
+func submitSpeechToText(ctx context.Context, params aiRequestParams, sess *AISession, req worker.SpeechToTextMultipartRequestBody) (*worker.TextResponse, error) {
+	var buf bytes.Buffer
+	mw, err := worker.NewSpeechToTextMultipartWriter(&buf, req)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := worker.NewClientWithResponses(sess.Transcoder(), worker.WithHTTPClient(httpClient))
+	if err != nil {
+		return nil, err
+	}
+
+	outPixels, err := common.CalculateAudioDuration(req.Audio)
+	if err != nil {
+		return nil, err
+	}
+	outPixels *= 1000 // Convert to milliseconds
+	setHeaders, balUpdate, err := prepareAIPayment(ctx, sess, outPixels)
+	if err != nil {
+		return nil, err
+	}
+	defer completeBalanceUpdate(sess.BroadcastSession, balUpdate)
+
+	start := time.Now()
+	resp, err := client.SpeechToTextWithBody(ctx, mw.FormDataContentType(), &buf, setHeaders)
+	took := time.Since(start)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, errors.New(string(data))
+	}
+
+	// We treat a response as "receiving change" where the change is the difference between the credit and debit for the update
+	if balUpdate != nil {
+		balUpdate.Status = ReceivedChange
+	}
+
+	var res worker.TextResponse
+	if err := json.Unmarshal(data, &res); err != nil {
+		return nil, err
+	}
+
+	// TODO: Refine this rough estimate in future iterations
+	sess.LatencyScore = took.Seconds() / float64(outPixels)
```

Comment:
I'd consider extracting all these "temp" conversions hacks (latency score, outPixels) to one place. Otherwise it may be hard later to reason about them.

Reply:
Added a linear issue to track this https://linear.app/livepeer-ai/issue/AI-505/refactor-latency-scoring-into-common-function